app/workers/slack_bot_ping_worker.rb

```diff
@@ -3,7 +3,24 @@ class SlackBotPingWorker
 
   sidekiq_options queue: :default, retry: 10
 
-  def perform(message, channel, username, icon_emoji)
-    SlackBot.ping(message, channel: channel, username: username, icon_emoji: icon_emoji)
+  def perform(slack_data = {})
+    # Sidekiq turns arguments into Strings so the Ruby keyword argument sorcery doesn't work here
+    # prevent any mismatch between String keys and Symbol keys
+    slack_data.symbolize_keys!
+
+    message = slack_data[:message]
+    channel = slack_data[:channel]
+    username = slack_data[:username]
+    icon_emoji = slack_data[:icon_emoji]
+
+    # Double check for any nil values
+    return unless message && channel && username && icon_emoji
```

Comment:
I know straightforward code is better than clever code but for future reference in case we add more parameters, this can be also done like:

```ruby
[1] pry(main)> data
=> {:message=>"yoo", :channel=>"yoo", :username=>"yooo", :icon_emoji=>""}
[2] pry(main)> %i[message channel username icon_emoji].all? { |key| data[key].present? }
=> false
[4] pry(main)> data = {:message=>"yoo", :channel=>"yoo", :username=>"yooo", :icon_emoji=>"yooo"}
=> {:message=>"yoo", :channel=>"yoo", :username=>"yooo", :icon_emoji=>"yooo"}
[4] pry(main)> %i[message channel username icon_emoji].all? { |key| data[key].present? }
=> true
```

Reply:
Generally agreed, just two comments:

1. Re-creating the same array is a bit of a waste so something like this is IMHO preferable
    ```ruby
    MANDATORY_KEYS = %i[message channel username icon_emoji].freeze
    ```
2. I prefer using `data.key?(key)` over `data[key]` in this context, as it avoids pitfalls like this one:
    ```ruby
    h = { foo: false, bar: 23 }
    [:foo, :bar].all? { |k| h[k] }
    #=> false
    [:foo, :bar].all? { |k| h.key?(k) }
    #=> true
    ```

Reply:
key will return true even if the key is blank or nil though ;)

Reply:
Yeah, nvm, the `.present?` part wasn't in my view port and I didn't notice the vertical scroll bar 🤦‍♂